app/services/user_service.py

```
@@ -0,0 +1,60 @@
+        hashed_password=hashed_password,
+        user_type=user_in.user_type,
+        is_active=True,
+    )
+    db.add(user)
+    try:
+        await db.commit()
+        await db.refresh(user)
+        return user
+    except IntegrityError:
+        await db.rollback()
+        raise ValueError("Email already registered")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)